plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSplitSource.java

```diff
@@ -356,4 +369,154 @@ public void testNullStatisticsMaps()
+                    public boolean isComplete()
+                    {
+                        return true;
+                    }
+
+                    @Override
+                    public boolean isAwaitable()
+                    {
+                        return false;
+                    }
+
+                    @Override
+                    public TupleDomain<ColumnHandle> getCurrentPredicate()
+                    {
+                        return TupleDomain.all();
+                    }
+                },
+                new Duration(2, SECONDS),
+                alwaysTrue(),
+                new TestingTypeManager(),
+                true);
+    }
+
+    private ImmutableList<IcebergSplit> getSplits(IcebergSplitSource splitSource)
+    {
+        ImmutableList.Builder<IcebergSplit> builder = ImmutableList.builder();
+        // set a large size to ensure all the data files had been handled
+        splitSource.getNextBatch(null, Integer.MAX_VALUE).whenComplete((splitBatch, ex) -> {
+            splitBatch.getSplits().forEach(split -> {
+                IcebergSplit icebergSplit = (IcebergSplit) split;
+                builder.add(icebergSplit);
+            });
+        });
+        return builder.build();
+    }
+
+    @Test
+    public void testSplitSourceWithSmallSplitSize()
+    {
+        String tableName = "test_default_split_size" + randomTableSuffix();
+        SchemaTableName schemaTableName = new SchemaTableName("tpch", tableName);
+        @Language("SQL") String createTableSql =
+                format("CREATE TABLE %s AS SELECT * FROM tpch.sf1.lineitem LIMIT 100000", tableName);
+        assertUpdate(createTableSql, 100000);
+
+        IcebergTableHandle tableHandle = new IcebergTableHandle(
+                schemaTableName.getSchemaName(),
+                schemaTableName.getTableName(),
+                TableType.DATA,
+                Optional.empty(),
+                TupleDomain.all(),
+                TupleDomain.all(),
+                ImmutableSet.of(),
+                Optional.empty());
+
+        ConnectorSession connectorSession = getSession().toConnectorSession(ICEBERG_CATALOG);
+        assertEquals(getSplitSize(connectorSession), DataSize.of(128, DataSize.Unit.MEGABYTE));
+        assertEquals(getSplitOpenFileCost(connectorSession), DataSize.of(4, DataSize.Unit.MEGABYTE));
+
+        Table icebergTable = loadIcebergTable(metastore, operationsProvider, connectorSession, schemaTableName);
+        CloseableIterable<FileScanTask> fileScanTasks = icebergTable.newScan().planFiles();
+
+        long totalDataSize = 0;
+        for (FileScanTask fileScanTask : fileScanTasks) {
+            totalDataSize += fileScanTask.length();
+
+            assertTrue(fileScanTask.length() <= getSplitSize(connectorSession).toBytes(),
+                    format("Split size %s is bigger than the default split size %s!",
+                            fileScanTask.length(), getSplitSize(connectorSession).toString()));
```

Comment:
We don't necessarily need to test the Iceberg library, just the Trino IcebergSplitSource code.

Reply:
Got it.